### PR TITLE
Tweak some defaults: `macro_names`, `operator_spaces`, `no_space`, `function_naming_convention`, `module_naming_convention`, `no_debug_call`

### LINKED ISSUE #234: Revisit defaults for rules
operator_spaces and no_space: we might want more operators on these ones...

Other rules: are all the ones we want as defaults in the rulesets? (should we add this info to the Wiki, that certain rules aren't executed by default?)

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
As per discussion under the issue this closes. Closes #234.

Documentation and migration guide are updated.

## Further considerations

Potentially best reviewed on a commit-by-commit basis: everything except the documentation updates is pretty isolated.

### CODE REVIEW

File: MIGRATION.md
Comment (paulo-ferraz-oliveira) [inline MIGRATION.md:1]: For all changes we target the equivalent 3.2.5 defaults (but any in the 3.x branch would be the same).

File: doc_rules/elvis_style/function_naming_convention.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ All functions should be named according to the regular expression provided.
 ## Options
 
 - `regex :: string()`.
-  - default: `"^([a-z][a-z0-9]*_?)*(_SUITE)?$"`.
+  - default: `"^[a-z](_?[a-z0-9]+)*$"`.
```
Comment (paulo-ferraz-oliveira) [inline doc_rules/elvis_style/function_naming_convention.md:10]: I mentioned it in the pull request, but here once again: does `_SUITE` actually make sense for the function name?

File: test/examples/fail_module_naming_1_convention_1.erl
Comment (paulo-ferraz-oliveira) [inline test/examples/fail_module_naming_1_convention_1.erl:1]: Not deleted, but renamed...

File: doc_rules/elvis_style/function_naming_convention.md (hunk 1 above)
Comment (elbrujohalcon) [reply doc_rules/elvis_style/function_naming_convention.md:10]: No, it does not.

**Review by elbrujohalcon — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
Strange that this is flagged as merged and closed 😄

**Pull request merged**